src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceCreationService.kt

```diff
@@ -243,6 +246,26 @@ class LicenceCreationService(
     return LicenceCreationResponse(createdLicence.id)
   }
 
+  fun determineLicenceKind(nomisRecord: PrisonerSearchPrisoner): LicenceKind {
+    val today = LocalDate.now()
+    val prrd = nomisRecord.postRecallReleaseDate
+
+    var kind: LicenceKind
+
+    if (prrd?.isAfter(today) == true && (nomisRecord.conditionalReleaseDate == null || prrd.isAfter(nomisRecord.conditionalReleaseDate))) {
+      kind = LicenceKind.PRRD
+    } else {
+      kind = LicenceKind.CRD
+    }
+
+    val hardstopDate = releaseDateService.getHardStopDate(nomisRecord.toSentenceDateHolder(null, kind))
+    if (hardstopDate != null && hardstopDate <= today) {
+      kind = LicenceKind.HARD_STOP
+    }
+
+    return kind
+  }
+
```

Comment:
this code looks familiar... did we do this in the case load service. could we call this from the case load service so we don't want to duplicate,... it dose not belong in the case load service it belongs here... I will go and check....
`caseloadService.determineLicenceKind(it)`
here it is, yes remove from case load service and use your code

Reply:
I can see you have removed I will shut up